### PR TITLE
make targets and prediction files for knight

### PR DESCRIPTION
**Pull request by mosheraboh**

### CODE REVIEW

File: fuse_examples/classification/knight/make_targets_file.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    lgr.info('KNIGHT: make targets file', {'attrs': ['bold', 'underline']})
+    lgr.info(f'targets_filename={os.path.abspath(output_filename)}', {'color': 'magenta'})
+
+    # Data
+    # read train/val splits file.
+    if isinstance(split, str): 
+        split=pd.read_pickle(split)
+        if isinstance(split, list):
+            # For this example, we use split 0 out of the the available cross validation splits
+            split = split[0]
```
Comment (alex-golts) [inline fuse_examples/classification/knight/make_targets_file.py:56]: do you want this assumption here (use of split 0 in particular)? this is meant to be a general file, no?

File: fuse_examples/classification/knight/make_predictions_file.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    lgr.info('KNIGHT: make predictions file in FuseMedML', {'attrs': ['bold', 'underline']})
+    lgr.info(f'predictions_filename={os.path.abspath(output_filename)}', {'color': 'magenta'})
+
+    # Data
+    # read train/val splits file.
+    if isinstance(split, str): 
+        split=pd.read_pickle(split)
+        if isinstance(split, list):
+            # For this example, we use split 0 out of the the available cross validation splits
+            split = split[0]
```
Comment (alex-golts) [inline fuse_examples/classification/knight/make_predictions_file.py:62]: same as below comment

File: fuse_examples/classification/knight/eval/eval.py [hunk 3]
```diff
@@ -137,49 +169,82 @@ def eval(task1_prediction_filename: str, task2_prediction_filename: str, target_
...
 
     # output
-    return decode_results(results, output_dir=output_dir)
+    return decode_results(results, output_dir=output_dir, task1=task1, task2=task2)
 
 
 if __name__ == "__main__":
     """
     Run evaluation:
     Usage: python eval.py <target_filename> <task1 prediction_filename> <task1 prediction_filename> <output dir>
```
Comment (alex-golts) [inline fuse_examples/classification/knight/eval/eval.py:232]: you wrote "task1 prediction filename" twice

File: fuse_examples/classification/knight/eval/eval.py [hunk 4]
```diff
@@ -1,3 +1,21 @@
+"""
```
Comment (alex-golts) [inline fuse_examples/classification/knight/eval/eval.py:1]: I just skimmed this file in very high level, sorry :(

File: fuse/data/dataset/dataset_default.py [hunk 5]
```diff
@@ -309,8 +309,10 @@ def get(self, index: Optional[Union[int, Hashable]], key: Optional[str] = None,
 
         # if key not specified return the all sample
         if key is None:
-            assert index != -1, 'get all samples is not supported when key = None'
-            return self.getitem(index)
+            if index is None:
```
[the remark this answers is not in the document]
Comment (mosheraboh) [reply fuse/data/dataset/dataset_default.py:312]: I didn't support this option so far (getting all data at once)

File: fuse_examples/classification/knight/README.md [hunk 6]
```diff
@@ -76,4 +76,26 @@ python eval.py example/example_targets.csv example/example_task1_predictions.csv
 ```    
 
 ## Baseline Implementation
+
```
[the remark this answers is not in the document]
Comment (mosheraboh) [reply fuse_examples/classification/knight/README.md:79]: Yes, when you'll move the baseline to public repo, you will merge the two readme files,

File: fuse_examples/classification/knight/README.md [hunk 7]
```diff
@@ -76,4 +76,26 @@ python eval.py example/example_targets.csv example/example_task1_predictions.csv
...
+TBD
+
+### **Model**
 TBD
+
+### **Implementation Details**
+TBD
+
+### **Make targets file for evaluation**
+Makes a targets file for the evaluation script using the PyTorch dataset specified in baseline implementation.
```
[the remark this answers is not in the document]
Comment (mosheraboh) [reply fuse_examples/classification/knight/README.md:90]: I will try to better clarify it. In high level, It's a file that holds the labels.

File: fuse_examples/classification/knight/README.md [hunk 8]
```diff
@@ -76,4 +76,26 @@ python eval.py example/example_targets.csv example/example_task1_predictions.csv
...
+
+You can either use the same train/validation split specified in baseline implementation or specify a different split.
+
+The script including additional details and documentation can be found in: 'fuse-med-ml/fuse_examples/classification/knight/make_targets_file.py'  
+
+### **Make targets file for evaluation**
```
[the remark this answers is not in the document]
Comment (mosheraboh) [reply fuse_examples/classification/knight/README.md:96]: My bad, thanks

File: fuse_examples/classification/knight/eval/eval.py (hunk 4 above)
Comment (mosheraboh) [reply fuse_examples/classification/knight/eval/eval.py:1]: NP, thanks

File: fuse_examples/classification/knight/make_predictions_file.py (hunk 2 above)
Comment (mosheraboh) [reply fuse_examples/classification/knight/make_predictions_file.py:62]: There is an option to provide the dictionary directly. So it's general.

**Pull request merged**